cmd/src/update.go

```diff
@@ -0,0 +1,156 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+
+	"github.com/sourcegraph/src-cli/internal/api"
+
+	"github.com/tj/go-update"
+	"github.com/tj/go-update/progress"
+	"github.com/tj/go-update/stores/github"
+)
+
+const (
+	// PrivateFileMode grants owner to read/write a file.
+	PrivateFileMode = 0600
+)
+
+func init() {
+	usage := `
+Examples:
+
+	Update and replace src-cli to the recommended version:
+
+		$ src update
+`
+
+	flagSet := flag.NewFlagSet("update", flag.ExitOnError)
+
+	var apiFlags = api.NewFlags(flagSet)
+
+	handler := func(args []string) error {
+		currentDirectory, err := os.Getwd()
+		if err != nil {
+			return err
+		}
+		err = isDirWritable(currentDirectory)
+		if err != nil {
+			runWithElevatedPrivilege()
```

Comment:
Two questions:

1. Why check whether the _current_ directory is writable? I would've assumed that we'd need to get the absolute path of the currently-executed `src` binary and then its dir?
2. Why run it with elevated privileges automatically? Couldn't we print an error message and tell the user something like "Hey, your `src` is located in `/usr/bin`, and the current user is lacking permissions to update it, please run with adminstrator/sudo permissions"?